### PR TITLE
fontface: add index parameter

### PR DESCRIPTION
**Pull request by ubitux**
Tested with the Iosevka TTC font

### CODE REVIEW

File: libnopegl/src/text_external.c [hunk 1]
```diff
@@ -60,9 +60,9 @@ static int load_font(struct text *text, const char *font_file)
     FT_Face ft_face = NULL;
     hb_font_t *hb_font = NULL;
 
-    FT_Error ft_error = FT_New_Face(text->ctx->ft_library, font_file, 0, &ft_face);
+    FT_Error ft_error = FT_New_Face(text->ctx->ft_library, font_file, face_index, &ft_face);
```
Comment (mbouron) [inline libnopegl/src/text_external.c:63]: I think we should sanitize face_index against negative values, see: https://freetype.org/freetype2/docs/reference/ft2-face_creation.html#ft_open_face.

File: libnopegl/src/node_fontface.c [hunk 2]
```diff
@@ -29,6 +29,8 @@ static const struct node_param fontface_params[] = {
     {"path",  NGLI_PARAM_TYPE_STR, OFFSET(path),
               .flags=NGLI_PARAM_FLAG_NON_NULL | NGLI_PARAM_FLAG_FILEPATH,
               .desc=NGLI_DOCSTRING("path to the font file")},
+    {"index", NGLI_PARAM_TYPE_I32, OFFSET(index),
+              .desc=NGLI_DOCSTRING("index of the face in the font file")},
```
Comment (mbouron) [inline libnopegl/src/node_fontface.c:33]: We should probably mention that the index is related to freetype `face_index` (https://freetype.org/freetype2/docs/reference/ft2-face_creation.html#ft_open_face).

File: libnopegl/src/text_external.c (hunk 1 above)
Comment (ubitux) [reply libnopegl/src/text_external.c:63]: Added a `[0,0xffff]` range check at fontface init

File: libnopegl/src/node_fontface.c (hunk 2 above)
Comment (ubitux) [reply libnopegl/src/node_fontface.c:33]: I'd rather not, freetype is an implementation detail. FT doc says "Bits 0-15 are the index of the face in the font file (starting with value 0)", so I believe using the description "index of the face in the font file" is consistent with that

**Review by mbouron — APPROVED** [review]

**Pull request merged**